### PR TITLE
model: pytorch: Custom exception when loss function is misspelled

### PR DESCRIPTION
**Pull request by nitesh585**
solved issue #910.

Exception message
before change - "AttributeError : 'NoneType' object has no attribute 'function' ".
after change - "LossFunctionNotFoundError : Loss function is not valid ".

### CODE REVIEW

File: model/pytorch/dffml_model_pytorch/pytorch_base.py [hunk 1]
```diff
@@ -63,6 +63,13 @@ def __post_init__(self):
                 map(self.clstype, self.classifications)
             )
 
+class LossFunctionNotFoundError(Exception):
+    '''
+        Exception raised when loss function is misspelled.
```
Comment (pdxjohnny) [inline model/pytorch/dffml_model_pytorch/pytorch_base.py:68]:
```suggestion
    Exception raised when loss function is misspelled.
```

File: model/pytorch/dffml_model_pytorch/pytorch_base.py [hunk 2]
```diff
@@ -63,6 +63,13 @@ def __post_init__(self):
...
+    '''
+    def __init__(self, message="Loss function is not valid!"):
+        self.message = message
+        super().__init__(self.message)
```
Comment (pdxjohnny) [inline model/pytorch/dffml_model_pytorch/pytorch_base.py:72]:
```suggestion
```

We don't need `__init__`. It is sufficient to just pass tje error message every time. If possible , we should report what the spelling the use gave that didn't work.

File: model/pytorch/dffml_model_pytorch/pytorch_base.py [hunk 3]
```diff
@@ -90,6 +97,9 @@ async def __aenter__(self):
 
         self.set_model_parameters()
 
+        if self.parent.config.loss is None:
+            raise LossFunctionNotFoundError()
```
Comment (pdxjohnny) [inline model/pytorch/dffml_model_pytorch/pytorch_base.py:101]:
```python
            raise LossFunctionNotFoundError(f"Loss function is not valid! Function name given was {<TODO: Figure out what variable would hold the original spelling>}")
```

**Review by pdxjohnny — CHANGES_REQUESTED** [review]

File: model/pytorch/dffml_model_pytorch/pytorch_base.py (hunk 3 above)
Comment (nitesh585) [reply model/pytorch/dffml_model_pytorch/pytorch_base.py:101]: Sure! Will try to figure out this.

**Comment by nitesh585** [comment]
Can't have access to loss function name given by the user in `./model/pytorch/dffml_model_pytorch/pytorch_base.py`.
The file where we can find that is   `./model/pytorch/dffml_model_pytorch/utils/utils.py`. So, I have changed it accordingly.

File: model/pytorch/dffml_model_pytorch/utils/utils.py [hunk 4]
```diff
@@ -76,6 +76,12 @@ def __len__(self):
         return len(self.data)
 
 
+class LossFunctionNotFoundError(Exception):
+    '''
+    Exception raised when Loss function is misspelled.
+    '''
+    pass
```
Comment (pdxjohnny) [inline model/pytorch/dffml_model_pytorch/utils/utils.py:83]:
```suggestion
```

Comment (nitesh585) [reply model/pytorch/dffml_model_pytorch/utils/utils.py]: @pdxjohnny Please review my latest commit. I have made all those changes.

**Comment by pdxjohnny** [comment]
Great work!
